### DISCUSSION TITLE
How to set fine grid number only at the interface not the whole model?

### DISCUSSION THREAD

**Discussion by abc-hy** [idaholab/moose#22495] [Q&A Modules: Phase field] [answered]
Hi everyone,

Does anyone know how to set a fine grid number only at the interface not the whole model? 
The following is the figure I got, I currently set the fine grid number for the whole model, how to only set a fine grid number at the interface? In this case I can save the computational cost.
![image](https://user-images.githubusercontent.com/63679626/197822782-ce74c904-a672-492a-96f4-6726383cdfc5.png)

Thank you very much,
Best,

**Comment by GiudGiud** [comment, marked as answer]
Hello

Yes you may add this mesh generator to refine only the boundary
https://mooseframework.inl.gov/source/meshgenerators/RefineSidesetGenerator.html

Guillaume

**Reply by abc-hy** [reply]
It seems that I need to label the interface coordinates? But, what if my interface is moving in the simulation? If it is moving, then how to determine the location?

**Reply by GiudGiud** [reply]
if it's moving then you should look at adaptivity instead. This will be run during the simulation

**Reply by abc-hy** [reply]
![image (1)](https://user-images.githubusercontent.com/63679626/197833445-ba6d0b0d-53c7-40aa-aac1-1762a45a0d25.png)
I am currently using this adaptivity block, which parameter should I change?

**Reply by GiudGiud** [reply]
You need to add an indicator and make sure it indicates the area you want to refine

**Reply by abc-hy** [reply]
how to add indicator?

**Reply by GiudGiud** [reply]
https://mooseframework.inl.gov/syntax/Adaptivity/index.html
https://mooseframework.inl.gov/syntax/Adaptivity/Indicators/index.html
https://mooseframework.inl.gov/syntax/Adaptivity/Markers/index.html
maybe this marker is exactly what you want
https://mooseframework.inl.gov/source/markers/BoundaryMarker.html
